CONTRIBUTING.md

```diff
@@ -0,0 +1,156 @@
+
+#### Virtual environments
+
+We strongly recommend using virtual environments when developing code in `dbt-trino`. We recommend creating this virtualenv
+in the root of the `dbt-trino` repository. To create a new virtualenv, run:
+```sh
+python3 -m venv env
+source env/bin/activate
+```
+
+This will create and activate a new Python virtual environment.
+
+#### Docker and `docker-compose`
+
+Docker and `docker-compose` are both used in testing. Specific instructions for you OS can be found [here](https://docs.docker.com/get-docker/).
+
+## Running `dbt-trino` in development
+
+### Installation
+
+First make sure that you set up your `virtualenv` as described in [Setting up an environment](#setting-up-an-environment).  Also ensure you have the latest version of pip installed with `pip install --upgrade pip`. Next, install `dbt-trino` (and its dependencies) with:
+
+```sh
+pip install -r dev-requirements.txt
+```
+
+When installed in this way, any changes you make to your local copy of the source code will be reflected immediately in your next `dbt` run.
+
+### Running `dbt-trino`
+
+With your virtualenv activated, the `dbt` script should point back to the source code you've cloned on your machine. You can verify this by running `which dbt`. This command should show you a path to an executable in your virtualenv.
+
+Configure your [profile](https://docs.getdbt.com/docs/configure-your-profile) as necessary to connect to your target databases. It may be a good idea to add a new profile pointing to a local Trino instance if appropriate.
+
+## Testing
+
+Once you're able to manually test that your code change is working as expected, it's important to run existing automated tests, as well as adding some new ones. These tests will ensure that:
+- Your code changes do not unexpectedly break other established functionality
+- Your code changes can handle all known edge cases
+- The functionality you're adding will _keep_ working in the future
+
+### Initial setup
+
+To be able to run the tests locally you will need a Trino or Starburst instance.
+
+```sh
+// to start Trino
+make start-trino
+// to start Starburst
+make start-starburst
+```
+
+### Test commands
+
+There are a few methods for running tests locally.
+
+#### Makefile
+
+There are multiple targets in the Makefile to run common test suites and code
+checks, most notably:
+
+```sh
+# Runs integration tests on Trino
+make dbt-trino-tests
+# Runs integration tests on Starburst
+make dbt-starburst-tests
+```
+> These make targets assume you have a local installation of a recent version of [`tox`](https://tox.readthedocs.io/en/latest/) for unit/integration testing and pre-commit for code quality checks,
+> unless you use choose a Docker container to run tests. Run `make help` for more info.
+
+#### `pre-commit`
```

Comment:
It's not used now, do you have plan to open a subsequent PR which adds `pre-commit`